src/adjoint.jl

```diff
@@ -45,17 +68,19 @@ function gradm(ex, mut = false)
     $adj
     @inline function ZygoteRules._pullback($cx, $f::$T, $(args...)) where $(Ts...)
       y, _back = adjoint(__context__, $f, $(argnames...))
-      $(mut ? nothing : :(back(::Nothing) = nothing))
-      back(Δ) = $gradtuple(_back(Δ))
+      $(mut ? nothing : :(back(::Union{Nothing,AbstractZero}) = Zero()))
+      #back(Δ) = $gradtuple(generic2diff(_back(diff2generic(Δ))))
+      back(Δ) = $gradtuple(generic2diff(_back(Δ)))
```

Comment:
This is the outstanding issue to be resolved: adding the `diff2generic` results in many warnings